src/dummy.rs

```diff
@@ -199,25 +199,45 @@ impl DummyVM {
     }
 
     fn read(&mut self) -> Result<String, EvalErrors> {
-        let mut buf = [0; 1024];
-        match self.stream.as_mut().unwrap().read(&mut buf) {
-            Result::Ok(n) => {
-                let s = std::str::from_utf8(&buf[..n])
-                    .expect("failed to parse the response, maybe the output is too long");
-                match s {
-                    "[Exception] SystemExit" => Err(EvalErrors::from(EvalError::system_exit())),
-                    "[Initialize]" => {
-                        self.compiler.initialize_generator();
-                        self.read()
-                    }
-                    _ => Ok(s.to_string()),
-                }
+        // Server Data Format:
+        // -------------------
+        // | size    | data
+        // -------------------
+        // | 2 bytes | n bytes
+        // -------------------
+
+        let mut size_buf = [0; 2];
+        match self.stream.as_mut().unwrap().read_exact(&mut size_buf) {
+            Result::Ok(()) => {}
+            Result::Err(err) => {
+                self.finish();
+                eprintln!("Read error: {err}");
+                process::exit(1);
             }
+        }
+
+        let data_size = u16::from_be_bytes(size_buf) as usize;
+        let mut data_buf = vec![0; data_size];
+
+        match self.stream.as_mut().unwrap().read_exact(&mut data_buf) {
+            Result::Ok(()) => {}
             Result::Err(err) => {
                 self.finish();
                 eprintln!("Read error: {err}");
                 process::exit(1);
             }
         }
+
+        let s = std::str::from_utf8(&data_buf)
+            .expect("failed to parse the response, maybe the output is too long");
+
+        if s.starts_with("[Exception] SystemExit") {
+            Err(EvalErrors::from(EvalError::system_exit()))
+        } else if let Some(remaing) = s.strip_prefix("[Initialize]") {
+            self.compiler.initialize_generator();
+            Ok(remaing.to_string())
```

Comment:
I'm not sure if this handling is correct, but the previous code returned the remaining content.